src/layouts/cc-validator/validation.js

```diff
@@ -117,39 +118,42 @@ function _checkKuhn() {
 
 function validateCard(acceptedCards, cardNumber) {
 	state.acceptedCards = acceptedCards;
-	state.cardNumber = cardNumber.replace(/\D/g,''); //strip extra characters
-	if (cardNumber.length === 0) {
+	state.cardNumber = cardNumber.trim(); //strip extra characters
```

Comment:
So we do need a way to remove spaces and dashes if someone types in their card like xxxx-xxxx-xxxx-xxxx. .replace(/\D/g,'') is too broad, but trim doesn't really accomplish all we need it to.